File: CS412-Algorithms/WCs/wc01/q2.py
import numpy as np
import matplotlib.pyplot as plt

n = np.arange(1, 1000001)  # Adjust as needed for 'large values' of n

log2_n = np.log2(n)  # Base 2
log7_n = np.log(n) / np.log(7)  # Change of base to 7
log10_n = np.log10(n)  # Base 10
log100_n = np.log(n) / np.log(100)  # Change of base to 100

# Set up the plot
plt.figure(figsize=(20, 10))

plt.plot(n, log2_n, label='Base 2', color='blue')
plt.plot(n, log7_n, label='Base 7', color='red')
plt.plot(n, log10_n, label='Base 10', color='green')
plt.plot(n, log100_n, label='Base 100', color='black')

# Add title and labels
plt.title('Logarithm functions with different bases')
plt.xlabel('n'); plt.ylabel('log(n)')
plt.legend(); 
plt.text(n[-1], log2_n[-1], r'$\log_2(n)$', fontsize=12, verticalalignment='bottom', horizontalalignment='right')
plt.text(n[-1], log7_n[-1], r'$\log_7(n)$', fontsize=12, verticalalignment='bottom', horizontalalignment='right')
plt.text(n[-1], log10_n[-1], r'$\log_{10}(n)$', fontsize=12, verticalalignment='bottom', horizontalalignment='right')
plt.text(n[-1], log100_n[-1], r'$\log_{100}(n)$', fontsize=12, verticalalignment='bottom', horizontalalignment='right')
plt.grid(True); plt.show()

File: CS412-Algorithms/WCs/wc01/q2.m
n = 1:1000000;  % Adjust as needed for 'large values' of n

% Calculate logarithms with different bases
log2_n = log2(n);  % Base 2
log7_n = log(n) / log(7);  % Change of base to 7
log10_n = log10(n);  % Base 10
log100_n = log(n) / log(100);  % Change of base to 100

% Plot the logarithm functions
figure;
plot(n, log2_n, 'b', 'LineWidth', 2);
hold on;
plot(n, log7_n, 'r', 'LineWidth', 2);
plot(n, log10_n, 'g', 'LineWidth', 2);
plot(n, log100_n, 'k', 'LineWidth', 2);

% Labels and legend
xlabel('n'); ylabel('log(n)');
title('Logarithm functions with different bases');
legend('Base 2', 'Base 7', 'Base 10', 'Base 100');
grid on; hold off;